bfvmm/src/exit_handler/test/test_exit_handler_intel_x64.cpp

```diff
@@ -2446,6 +2449,8 @@ exit_handler_intel_x64_ut::test_vmwrite_failure()
 
     RUN_UNITTEST_WITH_MOCKS(mocks, [&]
     {
-        EXPECT_EXCEPTION(eh->dispatch(), std::runtime_error);
+        auto f = [&] { eh->dispatch(); };
+        auto e = std::shared_ptr<std::exception>(new std::runtime_error("exit handler vmwrite failed"));
```

Comment:
I haven't looked at them all yet, but for the one highlighted, I can use std::bind...it looks like this:

```cpp
this->expect_exception(std::bind(&exit_handler_intel_x64::dispatch, std::move(eh)), std::make_shared<std::runtime_error>("error msg"));
```